test/Npgsql.PluginTests/JsonNetTests.cs

```diff
@@ -190,8 +195,37 @@ public class Bug3464Class
         public string? SomeString { get; set; }
     }
 
-    readonly NpgsqlDbType _npgsqlDbType;
-    readonly string _pgTypeName;
+    [Test]
+    [IssueLink("https://github.com/npgsql/npgsql/issues/4537")]
+    public async Task Write_jobject_array_without_npgsqldbtype()
+    {
+        // By default we map JObject to jsonb
+        if (!IsJsonb)
+            return;
+
+        await using var conn = await JsonDataSource.OpenConnectionAsync();
+        var tableName = await TestUtil.CreateTempTable(conn, "key SERIAL PRIMARY KEY, ingredients json[]");
```

Comment:
Is this inserting `jsonb` (because that's our default) into a `json` column? Is that an implicit conversion on the PG side?

Reply:
Yeah, from my understanding pg has an implicit cast from jsonb to json and the other way around.